### ISSUE TITLE
vbytes default size is 0

### ISSUE THREAD

**Issue by williballenthin** [vivisect/vstruct#3]
This issue can serve as a discussion point for the following behavior, which I was surprised by, and find inconsistent.

An uninitialized `vbytes` instance emits 0 bytes, while an uninitialized number instance emits a non-zero number of bytes. For example:

```
import vstruct.types as v_types

# initialized vbytes should have `size` bytes emitted
a = v_types.vbytes(size=8)
a.vsParse(b"\x00" * 8)
assert(len(a.vsEmit()) == 8)  # --> OK

# uninitialized vbytes should have `size` bytes emitted???
b = v_types.vbytes(size=8)
assert(len(b.vsEmit()) == 8)  # --> AssertionError: the size is 0

# initialized uint64 should have 8 bytes emitted
c = v_types.uint64()
c.vsParse(b"\x00" * 8)
assert(len(c.vsEmit()) == 8)  # --> OK

# uninitialized uint64 should have 8 bytes emitted
d = v_types.uint64()
assert(len(d.vsEmit()) == 8)  # --> OK
```

This surprised me while writing parsers because if I created a new instance of a VStruct that contained both number and vbytes members, I only had to explicitly initialize some of them. (Actually, I started with vstructs that only contained numbers, and didn't have to initialize the struct to emit a buffer of correct length. This changed when I added a vbytes member.)

I think the API would be more consistent if vbytes emitted `size` 0x00 bytes if its not been initialized; however, I can also understand the argument that structs should be initialized or undefined behavior results.

What are your thoughts?

**Comment by invisig0th** [comment]
Completely agreed.  x = vbytes(size=30).vsEmit() should totally give back 30 bytes of 0. ( before this conversation, i would have said it already worked that way :)  )  Purely an oversight.

**Comment by williballenthin** [comment]
closed in 461e3c7
